tests/cxx_gen.rs

```diff
@@ -32,3 +32,48 @@ fn test_impl_annotation() {
     let output = str::from_utf8(&generated.implementation).unwrap();
     assert!(output.contains("ANNOTATION void cxxbridge1$do_cpp_thing(::rust::Str foo)"));
 }
+
+const BRIDGE1: &str = r#"
+    #[cxx::bridge]
+    mod ffi {
+        unsafe extern "C++" {
+            type CppType;
+        }
+
+        extern "Rust" {
+            fn rust_method_cpp_context(self: Pin<&mut CppType>);
+        }
+    }
+"#;
+
+// Ensure that implementing a Rust method on a C++ type only causes generation
+// of the implementation.
+//
+// The header should be implemented in the C++ class definition and the Rust
+// implementation in the usual way.
```

Comment:
I don't understand what this middle paragraph means.

I will make a followup PR to reword this.

Reply:
#1571